pkg/sources/reconciler/azureservicebustopicsource/subscription.go

```diff
@@ -280,23 +195,16 @@ func recursErrMsg(errMsg string, err error) string {
 // isNotFound returns whether the given error indicates that some Azure
 // resource was not found.
 func isNotFound(err error) bool {
-	if dErr := (autorest.DetailedError{}); errors.As(err, &dErr) {
-		return dErr.StatusCode == http.StatusNotFound
-	}
-	return false
+	return errors.Is(err, servicebussdk.ErrNotFound{})
```

Comment:
NotFound error are also a pain. This code doesn't work because I should have used `As()` instead of `Is()`, but my real concern here is that even upstream is [hacking its way around those errors](https://github.com/Azure/azure-service-bus-go/blob/d0f6cc33729ca31fb8c75523f9bef00adb175e0a/subscription_manager.go#L340-L344).